Starter Kit installation/setup (#155)
AdobeDocs/commerce-extensibility: src/pages/app-development/starter-kit/project-setup.md

```json
{
  "product": ["commerce"],
  "customer": ["commerce", "backoffice"],
  "order": ["commerce", "backoffice"],
  "stock": ["commerce", "backoffice"],
  "shipment": ["commerce", "backoffice"]
}
```

### Execute the onboarding

Run the following command to generate the IO Event providers and the registrations for your Starter Kit project.

```bash
npm run onboard
```

The console displays the provider's IDs. The commerce instance and provider IDs will be used to configure your Commerce instance. You will need the backoffice provider ID to send the events to the App builder project.

```terminal
Process of On-Boarding done successfully: [
  {
    key: 'commerce',
    id: '<Commerce Provider ID>',
    instanceId: '<Instance ID of Commerce Provider',
    label: 'Commerce Provider'
  },
  {
    key: 'backoffice',
    id: '<Backoffice Provider ID>',
    instanceId: '<Instance ID of backoffice provider',
    label: 'Backoffice Provider'
  }
]
```

> Check your App developer console to confirm the registrations were created.

Reviewer: ```suggestion
Check your App in the Developer Console to confirm the registrations were created.
```

   ![General configuration](../../_images/events/general-configuration.png)

1. Select the server-to-server authorization method you implemented from the **Adobe I/O Authorization Type** menu. Adobe recommends using OAuth. JWT has been deprecated.

1. Copy the contents of the `<workspace-name>.json` file into the **Adobe I/O Workspace Configuration** field.

1. Enter a unique identifier in the **Adobe Commerce Instance ID** field. This value must contain English alphanumeric characters, underscores (_), and hyphens (-) only.

1. Click **Save Config**, but do not leave the page. The next section creates an event provider, which is necessary to complete the configuration.

1. Enable Commerce Eventing by setting **Enabled** to `Yes`.

   **Note**: You must enable cron so that Commerce can send events to the endpoint.

1. Enter the merchant's company name in the **Merchant ID** field. You must use alphanumeric and underscores only.

1. In the **Environment ID** field, enter a temporary name for your workspaces while in development mode. When you are ready for production, change this value to a permanent value, such as Production.

> 1. (Optional) By default, if an error occurs when Adobe Commerce attempts to send an event to Adobe I/O, Commerce retries a maximum of seven times. To change this value, uncheck the Use system value checkbox and set a new value in the **Maximum retries to send events** field.

Reviewer: ```suggestion
1. (Optional) By default, if an error occurs when Adobe Commerce attempts to send an event to Adobe I/O, Commerce retries a maximum of seven times. To change this value, uncheck the **Use system value** checkbox and set a new value in the **Maximum retries to send events** field.
```

> 1. (Optional) By default, Adobe Commerce runs a cron job (clean_event_data) every 24 hours that delete event data three days old. To change the number of days to retain event data, uncheck the Use system value checkbox and set a new value in the Event retention time (in days) field.

Reviewer: ```suggestion
1. (Optional) By default, Adobe Commerce runs a cron job (clean_event_data) every 24 hours that deletes event data three days old. To change the number of days to retain event data, uncheck the **Use system value** checkbox and set a new value in the **Event retention time (in days)** field.
```

1. Click **Save Config**.

### Subscribe to events in Adobe Commerce

Use the `bin/magento events:subscribe` command to subscribe to events, as described in [Subscribe and register events](../../events/configure-commerce.md#subscribe-and-register-events). The following table defines the events for each supported entity that you must subscribe to and lists the required fields, if applicable.

Entity | Event | Required fields
--- | --- | ---
Product | `com.adobe.commerce.observer.catalog_product_delete_commit_after` | Not applicable
Product | `com.adobe.commerce.observer.catalog_product_save_commit_after`   | `created_at`, `updated_at`
Customer | `com.adobe.commerce.observer.customer_save_commit_after` | `created_at`, `updated_at`
Customer | `com.adobe.commerce.observer.customer_delete_commit_after` | Not applicable
Customer group | `com.adobe.commerce.observer.customer_group_save_commit_after` | Not applicable
Customer group | `com.adobe.commerce.observer.customer_group_delete_commit_after` | `customer_group_code`
Order | `com.adobe.commerce.observer.sales_order_save_commit_after` | `created_at`, `updated_at`

> Stock | `com.adobe.commerce.observer.cataloginventory_stock_item_save_commit_after` | Not applicable

Reviewer: Shipment event seems to be missing?

1. From the Admin, go to **System** > Extensions > **Integrations**.

1. Click **Add New Integration**. The **New Integration** page displays.

   ![New integration](../../_images/starterkit/add-integration.png)

1. Enter a name for the integration in the **Name** field. Leave the other fields blank.

1. Click **API** under the Basic Settings menu to display the **Available APIs** screen. Change the value of the **Resource Access** drop-down menu to **All**.

1. Click **Integration Info** and enter your Admin password in the **Your Password** field. Click **Save** to return to the Integration page.

1. Click the **Activate** link in the grid.

  ![Activate link](../../_images/starterkit/activate-integration.png)

1. On the next page, click the **Allow** button to display the **Integration Tokens for Extensions** page.

   ![Integration tokens](../../_images/starterkit/secrets.png)

   You will need the integration details (consumer key, consumer secret, access token, and access token secret) to configure the Starter Kit. Copy these values to a safe place and click **Done**.

## Install Adobe I/O Events for Adobe Commerce (Commerce 2.4.4 and 2.4.5 only)

> If you are running Adobe Commerce 2.4.4 or 2.4.5, you must install modules to enable eventing, as described in [Install Adobe I/O Events for Adobe Commerce](../../events/installation.md). These modules are installed automatically in subsequent versions of Commerce.

Reviewer 2: ```suggestion
If you are running Adobe Commerce 2.4.4 or 2.4.5, you must install modules to enable eventing, as described in [Install Adobe I/O Events for Adobe Commerce](../events/installation.md). These modules are installed automatically in subsequent versions of Commerce.
```

## Starter Kit set-up and onboarding

You are now ready to download the Adobe Commerce Starter Kit and set up your development.

### Download and configure the project

> <!-- Note to reviewer: Will the enterprise-integration/starter-kit repo be mirrored? It looks like a private repo -->

Reviewer 3: During this beta phase, we are distributing the starter kit as a zip version of the git repo.
Once the beta phase is completed, the intention is to make the starter kit available in the marketplace.

   ```yaml
   application:
   runtimeManifest:
    packages:
      product-commerce:
        license: Apache-2.0
        actions:
          $include: ./actions/product/commerce/actions.config.yaml
    #  product-backoffice:
    #    license: Apache-2.0
    #    actions:
    #      $include: ./actions/product/external/actions.config.yaml
      customer-commerce:
        license: Apache-2.0
        actions:
          $include: ./actions/customer/commerce/actions.config.yaml
      customer-backoffice:
        license: Apache-2.0
        actions:
          $include: ./actions/customer/external/actions.config.yaml
    #  ...

### Deploy the project

Run the following command to deploy the project. The command deploys the runtime actions needed for the onboarding step:

```bash
aio app deploy
```

You can confirm the success of the deployment in the Adobe Developer Console by navigating to the Runtime section on your workspace.

![Adobe I/O Runtime actions](../../_images/starterkit/runtime-actions.png)

## Onboarding

> The onboarding process configuring event registrations and completing the eventing configuration in Adobe Commerce.
> 

Reviewer 3: Shouldn't present tense be used here?
```suggestion
The onboarding process configures event registrations and completes the eventing configuration in Adobe Commerce.

```

### Configure the event registrations

> By default, the `./onboarding/custom/registrations.json` config file creates all the registrations for all entities that are prtesen in the repo's `app.config.yaml` file. You can edit the `registrations.json` file to remove any Commerce or back office unneeded registrations. For example, the YAML file shown in the [Configure the project](#configure-the-project) section commented out the `product-backoffice` package. In this case, you must remove backoffice from the product entity:

Reviewer 3: Typo?
```suggestion
By default, the `./onboarding/custom/registrations.json` config file creates all the registrations for all entities that are present in the repo's `app.config.yaml` file. You can edit the `registrations.json` file to remove any Commerce or back office unneeded registrations. For example, the YAML file shown in the [Configure the project](#configure-the-project) section commented out the `product-backoffice` package. In this case, you must remove backoffice from the product entity:
```

### Subscribe to events in Adobe Commerce

> Use the `bin/magento events:subscribe` command to subscribe to events, as described in [Subscribe and register events](../../events/configure-commerce.md#subscribe-and-register-events). The following table defines the events for each supported entity that you must subscribe to and lists the required fields, if applicable.

Reviewer 4: ```
Use the `bin/magento events:subscribe` command to subscribe to events, as described in [Subscribe and register events](../../configure-commerce.md#subscribe-and-register-events). The following table defines the events for each supported entity that you must subscribe to and lists the required fields, if applicable.
```

Reviewer 4: remove the `events/`

Reviewer 4: or `[Subscribe and register events](../events/configure-commerce.md#subscribe-and-register-events)`
